tools/fastq_provider/fastq_provider.xml

```diff
@@ -1,78 +1,160 @@
+            <option value="auto" selected="true">auto</option>
+            <option value="ena">ENA</option>
+            <option value="sra">SRA file</option>
+            <option value="hca">Human Cell Atlas</option>
+        </param>
+        <param name="mode" argument="-p" type="select" label="Public or private" help="Public or private, defaults to public. Private usable by EBI staff only." optional="true">
             <option value="public" selected="true">public</option>
             <option value="private">private</option>
-	    </param>
-        <param name="library" argument="-l" type="text"  label="ENA library (ENA source files only), by default inferred from file name" help="library, by default inferred from file name. E.g. -l ERR1888646 " optional='true'/>
-        <param name="config_file" argument="-c" type="data" format="atlas-fastq-provider-config.sh" label="config file to override defaults" help="config file to override defaults" optional='true'/>
-        <param name="validate_only" argument="-v" type="boolean" label="validate only to check a source URI is valid, don't download" help="validate only, don't download" optional='true'/>
-        <param name="download_type" argument="-d" type="select" label="download type, fastq or SRA file" help="download type, fastq or srr" optional='true'>
-            <option value="fastq" selected="true">fastq</option>
-            <option value="srr">SRA</option>
-	    </param>
+        </param>
+        <param name="config_file" argument="-c" type="data" format="text" label="Config file to override defaults" help="Config file to override defaults." optional="true"/>
     </inputs>
     <outputs>
-        <data name="outfile" label="${tool.name} on ${on_string}: compressed fastq file" format="fastq.gz"  />
+        <data name="dl_from_path" label="Downloaded from path" format="fastqsanger.gz" from_work_dir="dl_from_path.fastq.gz">
+            <filter>fetch_type["script"] == "fetchFastq"</filter>
+        </data>
+        <data name="dl_from_lib_single" label="Downloaded library, single-end" format="fastqsanger.gz" from_work_dir="dl_from_lib_single.fastq.gz">
+            <filter>fetch_type["script"] == "fetchEnaLibraryFastqs" and fetch_type["sepe"] == "SINGLE"</filter>
+        </data>
+        <collection name="dl_from_lib_paired" type="list">
+          <filter>fetch_type["script"] == "fetchEnaLibraryFastqs" and fetch_type["sepe"] == "PAIRED"</filter>
+              <data name="dl_from_lib_pair_1" label="Downloaded library, pair read 1" format="fastqsanger.gz" from_work_dir="dl_from_lib_pair_1.fastq.gz"/>
+              <data name="dl_from_lib_pair_2" label="Downloaded library, pair read 2" format="fastqsanger.gz" from_work_dir="dl_from_lib_pair_2.fastq.gz"/>
```

Comment:
I think that there is a more specific way of handling paired end files in Galaxy as output.... like https://github.com/galaxyproject/tools-iuc/blob/e22821f51ae326b3696a9456cbc44f9982512a52/tools/fastp/fastp.xml#L232 or https://github.com/galaxyproject/tools-iuc/blob/e22821f51ae326b3696a9456cbc44f9982512a52/tools/fastp/fastp.xml#L328 (although maybe in your case you don't have a collection as output, but at least I suspect there is some paired type datatype). You might want to look for other example tools in that repo that deal with paired data.

Reply:
The example doesn't exactly apply because `type="data_collection" collection_type="paired"` is only for inputs, not for outputs (what we have in this wrapper is a paired output_collection).  But following the spirit of your suggestion, I have now changed the `output > collection` and corresponding `test > output_collection` type from `list` to `paired` to be more specific to a paired-end output (as in [this example](https://github.com/galaxyproject/tools-iuc/blob/5fcc308928ab2417b7e40227b27a3955f227649d/tools/read_it_and_keep/read-it-and-keep.xml#L147)).

Done in 542098133705df23c91aa41e393e3089debad8b5.